spec/models/course_spec.rb

```diff
@@ -143,119 +143,69 @@
   end
 
   describe '#upload_assignment_list' do
-    let!(:test_course) do
-      Course.where(name: 'TestCourse-2022-Fall', display_name: 'Test Course')
-            .first_or_create(name: 'TestCourse-2022-Fall', is_hidden: false,
-                             display_name: 'Test Course')
-    end
-    before(:each) do
-      test_course.assignments.destroy_all
-      expect(test_course.assignments.to_a).to eq([])
-    end
     context 'when file_format = \'csv\'' do
       context 'when the file contains no assignments' do
         it 'should not change the state of the database' do
           # get assignments associated with a course before uploading assignments
-          assignments_before_upload = test_course.assignments.to_ary
+          assignments_before_upload = course.assignments.to_a
 
-          test_course.upload_assignment_list('csv', [].to_csv)
+          course.upload_assignment_list('csv', [].to_csv)
 
           # reload the course so that we can see if the state of the course has changed in the database
-          test_course.reload
+          course.reload
 
-          assignments_after_upload = test_course.assignments.to_ary
+          assignments_after_upload = course.assignments.to_a
 
           # Expect there to be no stored assignments in the database
           expect(assignments_after_upload).to eq(assignments_before_upload)
         end
       end
       context 'when the file contains a single assignment' do
```

Comment:
(FIX/Simplification) 3 ------ https://github.com/MarkUsProject/Markus/pull/6281#discussion_r1008389661
(impacting tests where file has a single assignment)
- Changed the tests to no longer rely on whether or not the database was reloaded (previously on lines 181- 187)
        - Now just checks that message attribute has been changed.
-